### PR TITLE
feat(build): OAuth integration support for the action

### PR DESCRIPTION
**Pull request by kritikash18**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by kritikash18** [comment]
for testing runs, view: [https://github.com/kritikash18/aio-cloudmanager-create-execution-action/actions/runs/9527148266](https://github.com/kritikash18/aio-cloudmanager-create-execution-action/actions/runs/9527148266)

File: README.md [hunk 1]
```diff
@@ -49,17 +53,43 @@ This action can be configured two different ways -- either in combination with [
...
 * `CLIENTSECRET` - Client Secret for the project in the I/O Developer Console
-* `TECHNICALACCOUNTID` - Technical Account Email for the project in the I/O Developer Console
+* `TECHNICALACCOUNTID` - Technical Account ID for the project in the I/O Developer Console
+* `TECHNICALACCOUNTEMAIL` - Technical Account Email for the project in the I/O Developer Console (Only for OAuth integration)
 * `IMSORGID` - IMS Organization ID for the project in the I/O Developer Console
-* `SCOPES` - set to `ent_cloudmgr_sdk` **at minimum**
-* `KEY` - Private key for the project in the I/O Developer Console
+* `SCOPES` - set to `ent_cloudmgr_sdk` **at minimum** for JWT and `openid, AdobeID, read_organizations, additional_info.projectedProductContext, read_pc.dma_aem_ams` for OAuth
+* `KEY` - Private key for the project in the I/O Developer Console (Only for JWT integration)
+* `OAUTHENABLED` - set to true for OAuth-type integrations (required for OAuth integration; optional for JWT integration)
```
Comment (gjangidadobe) [inline README.md:93]: For JWT integration, shouldn't this param `OAUTHENABLED` be false?

File: action.yml [hunk 2]
```diff
@@ -11,17 +11,24 @@ inputs:
...
     required: false
   technicalAccountId:
-    description: 'When providing a JWT configuration, the technical account email from the Adobe Developer Console'
+    description: 'When providing a JWT/OAuth configuration, the technical account email from the Adobe Developer Console'
     required: false
   key:
     description: 'When providing a JWT configuration, the private key for the technical account'
     required: false
+  technicalAccountEmail:
+    description: 'When providing an OAuth configuration, the email ID for the technical account'
```
Comment (gjangidadobe) [inline action.yml]:
```suggestion
    description: 'When providing an JWT/OAuth configuration, the email ID for the technical account'
```

File: README.md (hunk 1 above)
Comment (kritikash18) [reply README.md:93]: if you check the `action.yml`, it is set to false by default

File: action.yml (hunk 2 above)
Comment (kritikash18) [reply action.yml]: it is not needed for JWT, only for OAuth - requirements of IMS library

File: action.yml [hunk 3]
```diff
@@ -11,17 +11,24 @@ inputs:
...
+    required: false
+  oauthEnabled:
+    description: 'When providing a JWT/OAuth configuration, the type of configuration enabled'
+    required: false
+    default: 'false'
```
Comment (gjangidadobe) [inline action.yml:31]: Got it

**Review by gjangidadobe — APPROVED** [review]
LGTM

**Pull request merged**